### PR TITLE
Feature/svg component

### PR DESCRIPTION
**Pull request by leandrosveltestudios**

### CODE REVIEW

File: src/Editor/Svg/SvgData.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { GalleryFileInfo } from '../../Gallery/Model';
+
+export type WhpptSvgData = {
+  svgString: string;
+  galleryItemId: string;
+  fileInfo?: GalleryFileInfo;
```
Comment (tjmkruger) [inline src/Editor/Svg/SvgData.tsx:6]: Don't think we need the fileInfo here. It lives on the gallery item and we have the gallery item id

Comment (leandrosveltestudios) [reply src/Editor/Svg/SvgData.tsx]: Done, left only  the `galleryItemId`, for the editor to pass the API.

File: src/ContentComponents/Svg/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  useEffect(() => {
+    if (data?.svg?.galleryItemId) {
+      setLoading(true);
+      api.gallery
+        .loadSvg(data?.svg?.galleryItemId)
+        .then(setSvgString)
+        .catch(err => setError(err.message || err))
+        .finally(() => setLoading(false));
+    }
+  }, [api.gallery, data]);
```
[the remark this answers is not in the document]
Comment (leandrosveltestudios) [reply src/ContentComponents/Svg/index.tsx]: Fixed.

File: src/styles/svgEditor.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
+.whppt-svg-editor-panel {
+  @apply flex flex-col gap-4;
+
+  &--empty {
+    @apply h-galleryItem w-galleryImage object-cover;
```
[the remark this answers is not in the document]
Comment (leandrosveltestudios) [reply src/styles/svgEditor.scss]: That was duplicated, yes. I just left width + aspect ratio. Might be good to extract this into it's own component on the UI improvements too.

File: src/ContentComponents/Svg/index.tsx [hunk 4]
```diff
@@ -12,20 +12,27 @@ export type SvgComponentData = {
...
+      .then(setSvgString)
+      .catch(err => setError(err.message || err))
+      .finally(() => setLoading(false));
+  }, [api.gallery, data?.svg?.galleryItemId]);
+
   useEffect(() => {
+    if (!created) return setCreated(true);
+
     if (data?.svg?.galleryItemId) {
       setLoading(true);
```
Comment (tjmkruger) [inline src/ContentComponents/Svg/index.tsx:32]: setLoading can move the the loadSvgString function. That way if we have to call it from anywehere else the loading state still gets set.

Comment (leandrosveltestudios) [reply src/ContentComponents/Svg/index.tsx]: done

**Pull request merged**